### DISCUSSION TITLE
Performance optimizations suggested by Lighthouse - how to implement?

### DISCUSSION THREAD

**Discussion by iGadget** [nextcloud/all-in-one#2161] [Questions] [answered]
As suggested on [the main page](https://github.com/nextcloud), I used Lighthouse to see if the performance of my local AIO server could be improved. Turns out, Lighthouse thinks it can be (it currently scores orange / 70):

1. Reduce unused Javascript
2. Enable text compression
3. Avoid enormous network payloads.

Apparently, Nextcloud sends over close to 20MB(!) of Javascript just to show the login page... Is this normal?
And how should I proceed to implement these suggestions?

**Comment by szaimen** [comment, marked as answer]
This is something that would need to be implemented in https://github.com/nextcloud/server

**Reply by iGadget** [reply]
Ok, but isn't that being done already? Since supposedly the main project is using Lighthouse.... or is it?

**Reply by szaimen** [reply]
Yes it is using Lighthouse but that doesnt mean that all recommendations are implemented yet.

**Reply by iGadget** [reply]
Ah, I see. So should I file this as a bug in /nextcloud/server then?

**Comment by pongraczi** [comment]
Just a report. (we use 26.0.4)

You are lucky, I got today 29 (in production) and 31 (fresh all-in-one) with the same recommendations.  

Filemanager is where we feel it annoying, the first appears is about 7-11seconds. Navigation after the first load of filemanager is pretty fast, but the first load and switch to other app and back are definitely slow.  

After the first start of the fresh aio installation this was so quick, my colleague measured about 1-2 seconds page load time. But since it degraded (1 week, even 2 days ago I turned off -> moved to stronger server with more resources -> started again) and the load time still significantly higher than in the first several days.

I start to dig deeper and follow several recommendations to see, what can I do in my side.


<details><summary>Installed Apps</summary>
<p>

```
root@naiodeb11:/var/lib# docker exec --user www-data nextcloud-aio-nextcloud php occ  app:list
Enabled:
  - activity: 2.18.0
  - admin_audit: 1.16.0
  - announcementcenter: 6.6.1
  - bruteforcesettings: 2.6.0
  - calendar: 4.4.3
  - calendar_resource_management: 0.5.0
  - circles: 26.0.0
  - cloud_federation_api: 1.9.0
  - comments: 1.16.0
  - contacts: 5.3.2
  - contactsinteraction: 1.7.0
  - dashboard: 7.6.0
  - dav: 1.25.0
  - deck: 1.9.2
  - drawio: 2.1.2
  - event_update_notification: 2.2.0
  - federatedfilesharing: 1.16.0
  - federation: 1.16.0
  - files: 1.21.1
  - files_accesscontrol: 1.16.0
  - files_antivirus: 5.2.1
  - files_bpm: 1.2.0
  - files_external: 1.18.0
  - files_fulltextsearch: 26.0.1
  - files_linkeditor: 1.1.16
  - files_mindmap: 0.0.28
  - files_pdfviewer: 2.7.0
  - files_rightclick: 1.5.0
  - files_sharing: 1.18.0
  - files_trashbin: 1.16.0
  - files_versions: 1.19.1
  - fileslibreofficeedit: 1.1.0
  - firstrunwizard: 2.15.0
  - forms: 3.3.1
  - fulltextsearch: 26.0.1
  - fulltextsearch_elasticsearch: 26.0.1
  - group_default_quota: 0.1.7
  - groupfolders: 14.0.3
  - groupquota: 0.1.11
  - guests: 2.5.0
  - impersonate: 1.13.1
  - integration_openai: 1.0.11
  - logreader: 2.11.0
  - lookup_server_connector: 1.14.0
  - mail: 3.2.5
  - maps: 1.1.0
  - metadata: 0.19.0
  - news: 22.0.0
  - nextcloud-aio: 0.4.0
  - nextcloud_announcements: 1.15.0
  - notes: 4.8.1
  - notifications: 2.14.0
  - notify_push: 0.6.3
  - oauth2: 1.14.1
  - password_policy: 1.16.0
  - passwords: 2023.8.30
  - photos: 2.2.0
  - polls: 5.2.0
  - privacy: 1.10.0
  - provisioning_api: 1.16.0
  - qownnotesapi: 23.6.0
  - quicknotes: 0.8.10
  - quota_warning: 1.17.0
  - recommendations: 1.5.0
  - related_resources: 1.1.0
  - richdocuments: 8.0.2
  - serverinfo: 1.16.0
  - settings: 1.8.0
  - sharebymail: 1.16.0
  - spreed: 16.0.5
  - support: 1.9.0
  - survey_client: 1.14.0
  - suspicious_login: 4.4.0
  - systemtags: 1.16.0
  - tasks: 0.15.0
  - terms_of_service: 2.2.0
  - text: 3.7.2
  - text_templates: 1.0.4
  - theming: 2.1.1
  - transfer: 0.6.1
  - twofactor_backupcodes: 1.15.0
  - twofactor_totp: 8.0.0
  - unroundedcorners: 1.0.9
  - user_ldap: 1.16.0
  - user_status: 1.6.0
  - viewer: 1.10.0
  - weather_status: 1.6.0
  - workflowengine: 2.8.0
Disabled:
  - encryption: 2.14.0
  - hibp: 0.1.0 (installed 0.1.0)
  - previewgenerator: 5.3.0 (installed 5.3.0)
```

</p>
</details> 

**Disabling brute force and rate limiting**
* There was no significant changes.

**Network inspection**
* It seems loading the filemanager app (reload or switch to Talk and back) again served from cache mostly, so, very low resources served by the server itself. Still, the rendering is about 5+ secs.  
Does it mean, there is a rendering issue, too?

**Reply by Dashrendar2507** [reply]
Have you made any discovery in that regard? I'm also experiencing roughly 7 second page loads with very few apps enabled and very good hardware.
